planner/core/logical_plan_builder.go

```diff
@@ -1994,6 +1998,44 @@ func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error) {
 	return result, nil
 }
 
+func (b *PlanBuilder) buildDataSourceFromView(dbName model.CIStr, tableInfo *model.TableInfo) (LogicalPlan, error) {
+	var (
+		selectNode        ast.StmtNode
+		selectLogicalPlan Plan
+		err               error
+	)
+	charset, collation := b.ctx.GetSessionVars().GetCharsetInfo()
+	selectNode, err = parser.New().ParseOneStmt(tableInfo.View.SelectStmt, charset, collation)
+	if err != nil {
+		return nil, err
+	}
+	selectLogicalPlan, err = b.Build(selectNode)
+	if err != nil {
+		return nil, err
+	}
+	projSchema := expression.NewSchema(make([]*expression.Column, 0, len(tableInfo.View.Cols))...)
+	for i := range tableInfo.View.Cols {
+		col := selectLogicalPlan.Schema().FindColumnByName(tableInfo.View.Cols[i].L)
+		if col == nil {
+			return nil, ErrViewInvalid.GenWithStackByArgs(dbName.O, tableInfo.Name.O)
+		}
+		projSchema.Append(&expression.Column{
+			UniqueID:    col.UniqueID,
```

Comment:
Can here just directly use the `col`?

Reply:
We need to change `ColName` and `OriginColName`  too.

Reply:
@winoros any suggestion for which test case function should I add, I was little confused?

Reply:
@winoros Add some plan test, PTAL

Reply:
I recommend you add some comment explaining why they can share the same `UniqueID`.